core/src/main/java/com/netease/arctic/io/writer/SortedPosDeleteWriter.java

```diff
@@ -114,7 +114,10 @@ public void delete(CharSequence path, long pos, T row) {
   }
 
   public List<DeleteFile> complete() throws IOException {
-    close();
+    io.doAs(() -> {
```

Comment:
I recommend wrapping the authentication logic around `SortedPosDeleteWriter#flushDeletes` because it will be called by other ways not only by `SortedPosDeleteWriter#complete`.

Reply:
> I recommend wrapping the authentication logic around `SortedPosDeleteWriter#flushDeletes` because it will be called by other ways not only by `SortedPosDeleteWriter#complete`.

OK， I think it will be better.